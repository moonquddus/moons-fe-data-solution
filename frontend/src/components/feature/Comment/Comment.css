.f-comment {
  padding-bottom: 8px;
  margin-bottom: 8px;
  border-bottom: solid 1px var(--calliper-color-grey);
}

.f-comment:last-of-type {
  border-bottom: none;
  margin-bottom: 0;
}

.f-comment__author {
  font-weight: var(--calliper-font-weight-bold);
  font-size: var(--calliper-font-size-xs);
}

.f-comment__text {
  margin: 0;
  font-size: var(--calliper-font-size-xs);
  line-height: var(--calliper-font-size-xs);
}