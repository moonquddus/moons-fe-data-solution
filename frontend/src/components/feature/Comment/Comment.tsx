import { Comment } from '../../../lib/model/comment'
import './Comment.css'

interface CommentProps {
  comment: Comment
}
/**
 * This can technically go inside element directory
 * As it's not actually doing anything, just a lovely minimal visual component
 */
function ChartComment({comment}: CommentProps) {
  const baseClass = 'f-comment'
  return (
    <li className={baseClass}>
      <div className={`${baseClass}__author`}>{comment.userName}</div>
      <p className={`${baseClass}__text`}>{comment.text}</p>
    </li>
  )
}

export default ChartComment
