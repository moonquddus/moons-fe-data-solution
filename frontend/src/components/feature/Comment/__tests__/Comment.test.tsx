import { render } from '@testing-library/react'
import { createComment } from '../../../../lib/testing/fixtures'
import Comment from '../Comment'

describe('rendering comment', () => {
  it('should correctly render the comment, given the props', () => {
    const comment = createComment()
    const wrapper = render(<Comment comment={comment} />)
    expect(wrapper.baseElement).toMatchSnapshot()
  })
})
