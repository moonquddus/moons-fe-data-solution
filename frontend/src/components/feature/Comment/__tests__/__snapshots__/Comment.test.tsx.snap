// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering comment should correctly render the comment, given the props 1`] = `
<body>
  <div>
    <li
      class="f-comment"
      data-testid="CommentItem"
    >
      <div
        class="f-comment__author"
      >
        Moon
      </div>
      <p
        class="f-comment__text"
      >
        Yoooooo check out this comment
      </p>
    </li>
  </div>
</body>
`;
